AnastasiyaTomasheva/JS/HW_14_and_exam/index.js

```diff
@@ -0,0 +1,736 @@
+
+let slider = document.createElement('div');
+    slider.classList.add('wrapper');
+    slider.innerHTML = `
+    <h1>Our friends</h1>
+    <div class="images-area">
+        <img src="./img/joom_.png" alt="image" class="firstImage">
+        <img src="./img/lstr_.png" alt="image">
+        <img src="./img/mydevice_.png" alt="image">
+        <img src="./img/oz_.png" alt="image">
+        <img src="./img/symbal_.png" alt="image">
+    </div>
+    <div class="buttons-area">
+        <div class="previous-btn">
+            <i class="bx bx-chevron-left"></i>
+        </div>
+        <div class="next-btn">
+            <i class="bx bx-chevron-right"></i>
+        </div>
+            <div class="pagination-area">
+        </div>
+    </div>
+    `
+
+    footer.before(slider);
+
+let imagesAreaImages = document.querySelectorAll('.images-area img'),
+    imagesAreaFirstImage = document.querySelector('.images-area .firstImage'),
+    previousBtn = document.querySelector('.previous-btn'),
+    nextBtn = document.querySelector('.next-btn'),
+    paginationArea = document.querySelector('.pagination-area'),
+    currentImageCount = 1,
+    sliderController;
+
+    previousBtn.addEventListener('click', previousImage);
+    nextBtn.addEventListener('click', nextImage);
+
+function previousImage(){
+    if(currentImageCount === 1){
+        return false
+    } else {
+        currentImageCount--
+        sliderController();
+    }
+}
+
+function nextImage() {
+
+    if(currentImageCount === imagesAreaImages.length){
+        return false;
+    }else{ 
+        currentImageCount++;
+        sliderController();
+    };
+};
+    
+(function createPaginationSpans(){
+    for(let i = 0; i < imagesAreaImages.length; i++){ 
+        let paginationSpan = document.createElement('span');
+            paginationArea.appendChild(paginationSpan)
+    };
+})();
+    
+(sliderController = function (){
+    let paginationCircls = document.querySelectorAll('.pagination-area span');
+        removeAllActive(paginationCircls);
+        activeButton();
+    let currentImageMinusOne = currentImageCount - 1;
+        paginationCircls[currentImageMinusOne].classList.add('active');
+        imagesAreaFirstImage.style.marginLeft = `-${1017 * currentImageMinusOne}px`;
+})();
+    
+function removeAllActive(targetElement){
+    for(let i = 0; i < targetElement.length; i++){
+        targetElement[i].classList.remove('active');
+    };
+};
+    
+function activeButton() {
+
+    currentImageCount === 1 ? 
+    previousBtn.classList.add('disabled') : 
+    previousBtn.classList.remove('disabled');
+        
+    currentImageCount === imagesAreaImages.length ? 
+    nextBtn.classList.add('disabled') : 
+    nextBtn.classList.remove('disabled');
+};
+    
+setInterval(() => {
+
+    if(currentImageCount != imagesAreaImages.length){
+        currentImageCount++;
+        sliderController();
+    }else{ 
+        currentImageCount = 1;
+        sliderController();
+    };
+}, 3000);
```

Comment:
very big file is 700 row. you should split into modules according to a functional parts